hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/ProtoConversionUtil.java

```diff
@@ -314,7 +351,16 @@ private Object convertObject(Schema schema, Object value) {
         case ENUM:
           return GenericData.get().createEnum(value.toString(), schema);
         case FIXED:
-          return GenericData.get().createFixed(null, ((GenericFixed) value).bytes(), schema);
+          if (value instanceof byte[]) {
+            return GenericData.get().createFixed(null, (byte[]) value, schema);
+          }
+          Object unsignedLongValue = value;
+          if (unsignedLongValue instanceof Message) {
+            // Unwrap UInt64Value
+            unsignedLongValue = getWrappedValue(unsignedLongValue);
+          }
+          // convert the long to its unsigned value
+          return DECIMAL_CONVERSION.toFixed(new BigDecimal(toUnsignedBigInteger((Long) unsignedLongValue)), schema, schema.getLogicalType());
```

Comment:
There's `BigDecimal.valueOf` we can use

Reply:
There is no public `valueOf` that takes in a BigInteger

Reply:
There's one that takes long

Reply:
Sync'd up offline, `BigDecimal.valueOf` treats longs as signed and there's no way to treat as unsigned. Keeping as is